backend/scheduler/tasks.py

```diff
@@ -85,16 +66,16 @@ def execute_pipeline_task(
         with tenant_context(tenant):
             workflow = Workflow.objects.get(id=workflow_id)
             logger.info(f"Executing workflow: {workflow}")
-            update_pipeline(pipepline_id, Pipeline.PipelineStatus.INPROGRESS)
+            PipelineProcessor.update_pipeline(
+                pipepline_id, Pipeline.PipelineStatus.INPROGRESS
+            )
             execution_response = WorkflowHelper.complete_execution(
                 workflow, execution_id, pipepline_id
             )
             logger.info(f"Execution response: {execution_response}")
-            update_pipeline(pipepline_id, Pipeline.PipelineStatus.SUCCESS)
         logger.info(f"Execution completed for pipeline: {name}")
     except Exception as e:
         logger.error(f"Failed to execute pipeline: {name}. Error: {e}")
-        update_pipeline(pipepline_id, Pipeline.PipelineStatus.FAILURE)
```

Comment:
@chandrasekharan-zipstack Can we keep the update for pipeline on failure case here ?

Reply:
@athul-rs it is valid to update the status here, however in order to avoid duplicating status updates and to cleanly update the same - I chose to move it to a single place. Currently its within a `workflow_execution` related file which itself is incorrect. But that refactor can come in later

Reply:
Okay, clear.